code/game/objects/items/shields.dm

```diff
@@ -255,3 +230,55 @@
 	if(attack_type & ATTACK_TYPE_PROJECTILE)
 		final_block_chance = 60 //Massive shield
 	return ..()
+
+
+/obj/item/shield/energy
+	name = "energy combat shield"
+	desc = "A shield that reflects almost all energy projectiles, but is useless against physical attacks. It can be retracted, expanded, and stored anywhere."
+	lefthand_file = 'icons/mob/inhands/equipment/shields_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/shields_righthand.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	attack_verb = list("shoved", "bashed")
+	throw_range = 5
+	force = 3
+	throwforce = 3
+	throw_speed = 3
+	var/base_icon_state = "eshield" // [base_icon_state]1 for expanded, [base_icon_state]0 for contracted
+	var/on_force = 10
+	var/on_throwforce = 8
+	var/on_throw_speed = 2
+	var/active = 0
+	var/clumsy_check = TRUE
+
+/obj/item/shield/energy/Initialize()
+	. = ..()
+	icon_state = "[base_icon_state]0"
```

Comment:
all icon_state changes should be in update_icon_state(). make this default to 0 don't make an initialize override over this. make update icon called during toggling or something.

Reply:
Oh this wasnt me. I just moved the E-shield down to its own thing...